engine/src/main/java/org/destinationsol/rubble/systems/RubbleCreationSystem.java

```diff
@@ -39,32 +51,122 @@
  */
 public class RubbleCreationSystem implements EventReceiver {
 
+    public static final float SIZE_TO_SHARD_COUNT = 13f;
+    public static final float MIN_SCALE = .07f;
+    public static final float MAX_SCALE = .12f;
+    private static final float MAX_SPD = 40f;
+
     @In
     private RubbleBuilder rubbleBuilder;
 
     @In
     private SolGame solGame;
 
-    //TODO once Shards are entities, this needs to be refactored to replace ShardBuilder
+    @In
+    private EntitySystemManager entitySystemManager;
 
     /**
-     * When an entity with a {@link CreatesRubbleOnDestruction} component is destroyed, this creates {@link Rubble}s where
+     * When an entity with a {@link CreatesRubbleOnDestruction} component is destroyed, this creates {@link Rubble} where
      * the entity was, unless the entity is in {@link Stasis}.
      */
     @ReceiveEvent(components = {CreatesRubbleOnDestruction.class, Position.class, Size.class})
     @Before(DestructionSystem.class)
     public EventResult onDeletion(DeletionEvent event, EntityRef entity) {
         if (!entity.hasComponent(Stasis.class)) {
-            Vector2 position = entity.getComponent(Position.class).get().position;
-            Vector2 velocity;
+            Velocity velocityComponent;
+            Angle angleComponent;
+            Size sizeComponent;
+            Position positionComponent;
+
             if (entity.hasComponent(Velocity.class)) {
-                velocity = entity.getComponent(Velocity.class).get().velocity;
+                velocityComponent = entity.getComponent(Velocity.class).get();
+            } else {
+                velocityComponent = new Velocity();
+                velocityComponent.velocity = new Vector2();
+            }
+
+
+            if (entity.hasComponent(Angle.class)) {
+                angleComponent = entity.getComponent(Angle.class).get();
+            } else {
+                angleComponent = new Angle();
+                angleComponent.setAngle(0f);
+            }
+
+
+            if (entity.hasComponent(Size.class)) {
```

Comment:
Since `@ReceiveEvent` requires that the entity have a Size component, the `if` statement is unnecessary. Same with the Position